### ISSUE TITLE
Can't convert bytes object to str implicitly

### ISSUE THREAD

**Issue by shongololo** [Toblerity/rtree#52]
I am getting the following error when trying to create a new disk based index:

<code>
File "/path/python3.4/site-packages/rtree/index.py", line 199, in __init__
    f = basename + "." + self.properties.idx_extension
TypeError: Can't convert 'bytes' object to str implicitly
</code>

**Comment by jbedorf** [comment]
I had the same problem when using Python34. I solved it by replacing this line:

https://github.com/Toblerity/rtree/blob/master/rtree/index.py#L199

with:
```
f = basename + "." + self.properties.idx_extension.decode("utf-8")
````
Not sure if it has any downsides, but at least the serialize function now works for me.

**Comment by hobu** [comment]
Fixed attempt in @9a7f3c7
